### PR TITLE
PyTorch model evaluation - `per_keypoint_evaluation` and `comparisonbodyparts`

### PR DESCRIPTION
**Pull request by n-poulsen**
This pull request implements the `per_keypoint_evaluation` and `comparisonbodyparts` parameters for PyTorch pose estimation models.

## Updates

- New `per_keypoint_evaluation ` argument for `deeplabcut.pose_estimation_pytorch.apis.evaluation.evaluate_network` to compute the errors (RMSE and mAP) based on a subset of bodyparts, as done for `tensorflow` models
- New `comparisonbodyparts` argument for `deeplabcut.pose_estimation_pytorch.apis.evaluation.evaluate_network` to compute RMSE per-bodypart and store the results in a CSV file, as done for `tensorflow` models
- Inclusion of unique bodypart RMSE in the "main" RMSE computation to match the `tensorflow` RMSE computation
- Additional tests to verify that RMSE computation is correct
- Updates the functional PyTorch tests to compute per-bodypart RMSE
- Update `docs/pytorch/user_guide.md` so that methods where parameters still need to be implemented are labeled orange and not green
- Update `docs/pytorch/user_guide.md` to reflect that `comparisonbodyparts` and `per_keypoint_evaluation` can now be used with PyTorch models

### CODE REVIEW

File: deeplabcut/pose_estimation_pytorch/apis/evaluate.py [hunk 1]
```diff
@@ -57,7 +56,6 @@ def predict(
     """Predicts poses on data contained in a loader
 
     Args:
-        pose_task: Whether the model is a top-down or bottom-up model
         pose_runner: The runner to use for pose estimation
         loader: The loader containing the data to predict poses on
         mode: {"train", "test"} The mode to predict on
```
Comment (maximpavliv) [inline deeplabcut/pose_estimation_pytorch/apis/evaluate.py:61]: Should we rephrase the detector_runner argument description (just below) as there is no task anymore?

File: deeplabcut/pose_estimation_pytorch/apis/evaluate.py [hunk 2]
```diff
@@ -97,61 +95,95 @@ def predict(
...
+    per_keypoint_evaluation: bool = False,
     pcutoff: float = 1,
 ) -> tuple[dict[str, float], dict[str, dict[str, np.ndarray]]]:
     """
     Args:
-        pose_task: Whether to run top-down or bottom-up
         pose_runner: The runner for pose estimation
         loader: The loader containing the data to evaluate
         mode: Either 'train' or 'test'
         detector_runner: If task == 'TD', a detector can be given to compute bounding
```
Comment (maximpavliv) [inline deeplabcut/pose_estimation_pytorch/apis/evaluate.py]: Should we rephrase this as there is no task anymore?

**Review by maximpavliv — APPROVED** [review]
Beside the two comments regarding documentation - this looks good to me!

File: deeplabcut/pose_estimation_pytorch/apis/evaluate.py (hunk 1 above)
Comment (n-poulsen) [reply deeplabcut/pose_estimation_pytorch/apis/evaluate.py:61]: done!
